src/julia_threads.h

```diff
@@ -428,8 +442,10 @@ static inline void jl_mutex_lock_nogc(jl_mutex_t *lock)
 
 static inline void jl_mutex_lock(jl_mutex_t *lock)
 {
+    JL_SIGATOMIC_BEGIN();
```

Comment:
but wouldn't that mean the finalizers are being run with sigatomic set? maybe we should address that in the finalizer runner though.